web/src/actions/user.test.js

```diff
@@ -0,0 +1,119 @@
+      type: actions.GET_USER_SUCCESS,
+      data
+    });
+  });
+
+  it('failUser should create GET_USER_FAILURE action', () => {
+    expect(actions.failUser('foo')).toEqual({
+      type: actions.GET_USER_FAILURE,
+      error: 'foo'
+    });
+  });
+
+  it('requestUserUpdate should create UPDATE_USER_REQUEST action', () => {
+    expect(actions.requestUserUpdate()).toEqual({
+      type: actions.UPDATE_USER_REQUEST
+    });
+  });
+
+  it('receiveUserUpdate should create UPDATE_USER_SUCCESS action', () => {
+    const data = { name: 'foo' };
+    expect(actions.receiveUserUpdate(data)).toEqual({
+      type: actions.UPDATE_USER_SUCCESS,
+      data
+    });
+  });
+
+  it('failUserUpdate should create GET_USER_FAILURE action', () => {
+    expect(actions.failUserUpdate('foo')).toEqual({
+      type: actions.UPDATE_USER_FAILURE,
+      error: 'foo'
+    });
+  });
+
+  describe('fetchUser (async)', () => {
+    afterEach(() => {
+      fetchMock.reset();
+    });
+
+    it('creates GET_USER_SUCCESS after successful user fetch', () => {
+      const store = mockStore({});
+      fetchMock.onGet().reply(200, { foo: 'bar' });
+
+      const expectedActions = [
+        { type: actions.GET_USER_REQUEST },
+        { type: actions.GET_USER_SUCCESS, data: { foo: 'bar' } }
+      ];
+
+      return store.dispatch(actions.fetchUser()).then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+      });
+    });
+
+    it('creates GET_USER_FAILURE after unsuccessful user fetch', () => {
+      const store = mockStore({});
+      fetchMock.onGet().reply(403, 'foo');
+
+      const expectedActions = [
+        { type: actions.GET_USER_REQUEST },
+        { type: actions.GET_USER_FAILURE, error: 'foo' }
+      ];
+
+      return store.dispatch(actions.fetchUser()).then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+      });
+    });
+  });
+
+  describe('updateUser (async)', () => {
+    afterEach(() => {
+      fetchMock.reset();
+    });
+
+    it('creates UPDATE_USER_SUCCESS after successful user fetch', () => {
+      const store = mockStore({});
+      fetchMock.onPut().reply(200, { foo: 'bar' });
+
+      const expectedActions = [
+        { type: actions.UPDATE_USER_REQUEST },
+        { type: actions.UPDATE_USER_SUCCESS, data: { foo: 'bar' } }
+      ];
+
+      return store.dispatch(actions.updateUser()).then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+      });
+    });
+
+    it('creates UPDATE_USER_FAILURE after unsuccessful user fetch', () => {
+      const store = mockStore({});
+      fetchMock.onPut().reply(403, 'foo');
+
+      const expectedActions = [
+        { type: actions.UPDATE_USER_REQUEST },
+        { type: actions.UPDATE_USER_FAILURE, error: 'foo' }
+      ];
+
+      return store.dispatch(actions.updateUser()).then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+      });
+    });
+  });
+});
```

Comment:
No test for `fetchUserDataIfNeeded`?  😄

I think it'd only need to check that the `GET_USER_REQUEST` action is dispatched or not, depending on the state, since that's the only unique logic.  For the correct functioning of everything else, we can rely on the `GET_USER_SUCCESS` and `GET_USER_FAILURE` tests up above.

Reply:
agreed, needs to be added